cpp/oneapi/dal/detail/common.hpp

```diff
@@ -72,4 +72,76 @@ Object make_from_pointer(typename Object::pimpl::element_type* pointer) {
     return pimpl_accessor().template make_from_pointer<Object>(pointer);
 }
 
+constexpr std::int64_t get_data_type_size(data_type t) {
+    if (t == data_type::float32) {
+        return sizeof(float);
+    }
+    else if (t == data_type::float64) {
+        return sizeof(double);
+    }
+    else if (t == data_type::int32) {
+        return sizeof(int32_t);
+    }
+    else if (t == data_type::int64) {
+        return sizeof(int64_t);
+    }
+    else if (t == data_type::uint32) {
+        return sizeof(uint32_t);
+    }
+    else if (t == data_type::uint64) {
+        return sizeof(uint64_t);
+    }
+    return 0;
```

Comment:
Shouldn't it cause compile or runtime error?

Reply:
it should.